src/gui/updater/ocupdater.cpp

```diff
@@ -45,7 +46,30 @@ UpdaterScheduler::UpdaterScheduler(QObject *parent)
         connect(updater, &OCUpdater::newUpdateAvailable,
             this, &UpdaterScheduler::updaterAnnouncement);
 
-        connect(updater, &OCUpdater::requestRestart, this, &UpdaterScheduler::requestRestart);
+        connect(updater, &OCUpdater::requestRestart, this, [this]() {
+            // prevent dialog from being displayed twice (rather unlikely, but it won't hurt)
+            if (_updateFinishedDialog == nullptr) {
+                _updateFinishedDialog = new UpdateFinishedDialog(_parentWidget);
+                _updateFinishedDialog->setAttribute(Qt::WA_DeleteOnClose);
+                _updateFinishedDialog->show();
+                ownCloudGui::raiseDialog(_updateFinishedDialog);
```

Comment:
Dialogs passed to raiseDialog need to be parented by settingsDialog so passing it as QWidget is shadowing this.

Reply:
Ok. But I wonder why `raiseDialog` doesn't enforce the use with `QDialog`s then (and `QMainWindow`). We should assert this or provide two signatures.

Reply:
https://github.com/owncloud/client/blob/769fa6f9fc34d8084e87fb0669ec413cec0af667/src/gui/owncloudgui.cpp#L985-L985

Reply:
I wonder, how was this assert rendered unusable while passing the variable as a `QWidget`? I mean, the code seemed to work and the assert did not trigger.